lib/elixir/src/elixir_errors.erl

```diff
@@ -57,6 +61,125 @@ emit_diagnostic(Severity, Position, File, Message, Stacktrace) ->
 
   ok.
 
+% --------- Fancy Diagnostics
+
+fancy_exception(File, LineNumber, Column, Description, Snippet) ->
```

Comment:
Nitpick: can we replace the word "fancy" by "snippet"?

Reply:
What would we call the function head where there's no snippet? Still fancy_exception or we could have `format_exception` for both?

Reply:
@viniciusmuller when there is no snippet, we still try to retrieve one and only then fallback. So I would go with either `format_snippet` or `format_location`. :)

Reply:
I remember you said in the live yesterday that when there's no snippet it means something weird happen and it was omitted, so we were not going try and fetch the line because it might point to the EOF line, for example. Or did I get it wrong?

Reply:
Yes, sorry, you are correct for syntax/token missing error! But I think we may need to read it later, for example, when the user calls `IO.warn`, right?

Reply:
But i am fine with not reading it now and removing the code, and reading later when we need it.